### DISCUSSION TITLE
Reuse json_annotation

### DISCUSSION THREAD

**Discussion by jonasbark** [simc/crimson#10] [Ideas]
Hi!

Exciting project - I always saw JSON parsing as a bottleneck in e.g. Flutter Web apps that can't create new isolates.

My idea is the following: Instead of creating new annotations one might be able to reuse the ones provided by the json_annotation package:
https://github.com/google/json_serializable.dart/tree/master/json_annotation/lib/src

The advantage would be the very simple interchangeability with json_serializable and the reusing of its documentation. Not sure how much flexibility crimson needs additionally.

**Comment by simc** [comment]
That's a nice idea, I'll look into it 🙂
